extension/background.ts

```diff
@@ -3,34 +3,32 @@ import { config } from './config.js'
 
 const FETCH_TIMEOUT_MS = 30000
 
-const sendMessage = (
+function sendMessage(
   port: chrome.runtime.Port,
   message: ExtensionMessage
-): void => {
+): void {
   port.postMessage(message)
 }
 
 chrome.runtime.onConnectExternal.addListener(port => {
   if (port.sender?.origin === config.HOST) {
     sendMessage(port, { type: 'connected' })
     port.onMessage.addListener(async (message: AppMessage) => {
-      if (message.url) {
-        await fetch(message.url, {
+      try {
+        let response = await fetch(message.url, {
```

Comment:
By using `await` everywhere we are having better readability in contrast with mixing `await` and `.then()`